dask_cuda/dgx.py

```diff
@@ -0,0 +1,41 @@
+import os
+
+from dask.distributed import Nanny, SpecCluster, Scheduler
+from distributed.bokeh.scheduler import BokehScheduler
+
+from .local_cuda_cluster import cuda_visible_devices
+
+
+def DGX(**kwargs):
+    gpus = map(
+        int, os.environ.get("CUDA_VISIBLE_DEVICES", "0,1,2,3,4,5,6,7").split(",")
+    )
+
+    spec = {
+        i: {
+            "cls": Nanny,
+            "options": {
+                "env": {
+                    "CUDA_VISIBLE_DEVICES": cuda_visible_devices(i, range(8)),
+                    # 'UCX_NET_DEVICES': 'mlx5_%d:1' % (i // 2)
```

Comment:
When I uncomment this line, things break.

Reply:
Does each `i ` correspond to a distinct worker?  If so, it makes sense to me that the `UCX_NET_DEVICES` line is causing you problems.  I think you would need to list the HCA:port combo for the worker, as well as the scheduler (and maybe even other workers if there is worker-worker communication)

Reply:
There is definitely worker-worker communication.  All workers will talk to each other.

What should this line be?

Reply:
Also, if you're interested and have time, I would encourage you to play with this branch and see if you can get the benchmark below working well.  I'd be happy to help in any way with startup difficulties.

Reply:
I will experiment a bit and keep you updated :)

Reply:
@mrocklin When you uncomment, you're asking different ucx processes to use different IB devices. Now that I think about this, I think during endpoint creation and connection establishment, if the states aren't consistent across processes there could be potential issues. I know this is contrary to the approach we discussed recently and apologize if I mislead you into doing this. Ideally, with the patches I have in `1ad5b17/ucx-cuda` you don't have to manually pick the IB devices but based on the GPU context selected, the closest HCA gets picked for transfers. 

Regardless, can you paste the error here?

I'll also give this is a shot today soon.

Reply:
I believe that @rjzamora is getting an error similar to what we found in https://github.com/Akshay-Venkatesh/ucx-py/issues/102

Reply:
> I'll also give this is a shot today soon.

That would be very welcome :)